plugins/filter/validate.py

```diff
@@ -0,0 +1,154 @@
+          this case the value I(ansible.utils.jsonschema) represents the engine to be use for data valdiation.
+          If the value is not provided the default value that is I(ansible.uitls.jsonschema) will be used.
+        - The value should be in fully qualified collection name format that is
+          I(<org-name>.<collection-name>.<validator-plugin-name>).
+        default: ansible.utils.jsonschema
+    notes:
+    - For the type of options C(data) and C(criteria) refer the individual C(validate) plugin
+      documentation that is represented in the value of C(engine) option.
+    - For additional plugin configuration options refer the individual C(validate) plugin
+      documentation that is represented by the value of C(engine) option.
+    - The plugin configuration option can be either passed as I(key=value) pairs within filter plugin
+      or environment variables.
+    - The precedence of the C(validate) plugin configurable option is the variable passed within filter plugin
+      as I(key=value) pairs followed by the environment variables.
+"""
+
+EXAMPLES = r"""
+- name: set facts for data and criteria
+  set_fact:
+    data: "{{ lookup('file', './validate/data/show_interfaces_iosxr.json')}}"
+    criteria: "{{ lookup('file', './validate/criteria/jsonschema/show_interfaces_iosxr.json')}}"
+
+- name: validate data in json format using jsonschema with by passing plugin configuration variable as key/value pairs
+  ansible.builtin.set_fact:
+    data_validilty: "{{ data|ansible.utils.validate(criteria, engine='ansible.utils.jsonschema', draft='draft7') }}"
+"""
+
+RETURN = """
+  _raw:
+    description:
+      - If data is valid returns empty list
+      - If data is invalid returns list of errors in data
+"""
+
+from copy import deepcopy
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible_collections.ansible.utils.plugins.module_utils.validate.base import (
+    load_validator,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    to_list,
+)
+from ansible_collections.ansible.utils.plugins.module_utils.common.argspec_validate import (
+    check_argspec,
+)
+
+try:
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+ARGSPEC_CONDITIONALS = {}
+
+
+def validate(*args, **kwargs):
+    if len(args) < 2:
+        raise AnsibleError(
+            "Missing either 'data' or 'criteria' value in filter input,"
+            "refer ansible.utils.validate filter plugin documentation for detials"
+        )
+
+    params = {"data": args[0], "criteria": args[1]}
+    if kwargs.get("engine"):
+        params.update({"engine": kwargs["engine"]})
+
+    argspec = deepcopy(DOCUMENTATION)
+    argspec_obj = yaml.load(argspec, SafeLoader)
+
+    argspec_result, updated_params = check_argspec(
```

Comment:
pure curiosity why not valid, errors, updated_params = check_argspec and skip the gets

Reply:
Good point. Fixed it